hibernate5/src/main/java/com/baeldung/hibernate/HibernateMultiTenantUtil.java

```diff
@@ -0,0 +1,95 @@
+package com.baeldung.hibernate;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.Environment;
+import org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl;
+import org.hibernate.engine.jdbc.connections.spi.ConnectionProvider;
+import org.hibernate.service.ServiceRegistry;
+
+import com.baeldung.hibernate.pojo.Suppliers;
+
+public class HibernateMultiTenantUtil {
+    private static SessionFactory sessionFactory;
+    private static Map<String, ConnectionProvider> connectionProviderMap = new HashMap<>();
+    private static final String[] tenantDBNames = { "mydb1","mydb2"};
+
+    public static SessionFactory getSessionFactory() throws UnsupportedTenancyException {
+        if (sessionFactory == null) {
+            Configuration configuration = new Configuration().configure();
+            ServiceRegistry serviceRegistry = configureServiceRegistry(configuration);
+            sessionFactory = makeSessionFactory (serviceRegistry);
+//            sessionFactory = configuration.buildSessionFactory(serviceRegistry);
+            
+            
+        }
+        return sessionFactory;
+    }
+
+    private static SessionFactory makeSessionFactory(ServiceRegistry serviceRegistry) {
+        MetadataSources metadataSources = new MetadataSources( serviceRegistry );
+        for(Class annotatedClasses : getAnnotatedClasses()) {
```

Comment:
@KevinGilmore we can do that but will have to convert array results given by getAnnotedClasses() to list and then use stream. So I thought to keep it as it is. What do you feel ?